domain/cloudimagemetadata/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package bootstrap
+
+import (
+	"context"
+
+	"github.com/juju/clock"
+
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/domain/cloudimagemetadata"
+	"github.com/juju/juju/domain/cloudimagemetadata/state"
+	"github.com/juju/juju/environs/imagemetadata"
+	"github.com/juju/juju/environs/simplestreams"
+	internaldatabase "github.com/juju/juju/internal/database"
+)
+
+func InitCustomImageMetadata(clock clock.Clock, defaultStream string, metadata []*imagemetadata.ImageMetadata) internaldatabase.BootstrapOpt {
```

Comment:
Comment me.

nit: Also `Add` vs `Init`, think the former sounds better, but not enough for me to block this PR.